File: challenges/src/test/java/leftJoin/LeftJoinTest.java
package leftJoin;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;

import java.util.HashMap;

public class LeftJoinTest {

    HashMap<String, String> synonymHash = new HashMap<>();
    HashMap<String, String> antonymHash = new HashMap<>();
    @Before
    public void hashMapCreate(){
        synonymHash.put("fond", "enamored");
        synonymHash.put("wrath", "angered");
        synonymHash.put("diligent", "employed");
        synonymHash.put("outfit", "garb");
        synonymHash.put("guide", "usher");

        antonymHash.put("fond", "averse");
        antonymHash.put("wrath", "delight");
        antonymHash.put("diligent", "idle");
        antonymHash.put("guide", "follow");
        antonymHash.put("flow", "jam");

    }
    @Test public void leftJoinTest(){
        LeftJoin joinTest = new LeftJoin();
        String expected = "[[diligent, employed, idle][outfit, garb, null][wrath, angered, delight][guide, usher, follow][fond, enamored, averse]]";
        HashMap<String,String> empty = new HashMap<>();

        //Test on example
        joinTest.leftJoin(synonymHash, antonymHash);
        assertEquals(expected,joinTest.toString());

        // Test that method can handle a empty second hash
        joinTest.leftJoin(synonymHash, empty);
        expected = "[[diligent, employed, null][outfit, garb, null][wrath, angered, null][guide, usher, null][fond, enamored, null]]";
        assertEquals(expected, joinTest.toString());

        // Test that a empty array is return if first hash is empty
        joinTest.leftJoin(empty, synonymHash);
        expected = "[]";
        assertEquals(expected, joinTest.toString());
    }
}

File: dotnet/XUnitTestProject1/LeftJoinTest.cs
using System;
using Xunit;
using Implementations;
using LeftJoin;

namespace XUnitTestProject1
{
  public class LeftJoinTest
  {
    [Fact]
    public void HappyPath()
    {
      HashMap synonym = new HashMap(20);
      synonym.Add("fond", "enamored");
      synonym.Add("wrath", "anger");
      synonym.Add("diligent", "employed");
      synonym.Add("outfit", "garb");
      synonym.Add("guide", "usher");

      HashMap antonym = new HashMap(20);
      antonym.Add("fond", "averse");
      antonym.Add("wrath", "delight");
      antonym.Add("diligent", "idle");
      antonym.Add("guide", "follow");
      antonym.Add("flow", "jam");

      string[][] test = Program.LeftJoin(synonym, antonym);
      string[][] expected = new string[5][];
      expected[0] = new string[] { "wrath", "anger", "delight" };
      expected[1] = new string[] { "diligent", "employed", "idle" };
      expected[2] = new string[] { "outfit", "garb", "NULL" };
      expected[3] = new string[] { "guide", "usher", "follow" };
      expected[4] = new string[] { "fond", "enamored", "averse" };
      
      int counter = 0;
      
      foreach (string[] strArr in test)
      {
        if (strArr != null)
        {
          for(int i = 0; i < strArr.Length; i++)
          {
            Assert.Equal(strArr[i], expected[counter][i]);
          }
          counter++;
        }
      }
    }

  }
}
